e2e-test/src/test/java/org/alfresco/hxi_connector/e2e_test/CreateNodeE2eTest.java

```diff
@@ -0,0 +1,162 @@
+
+@Slf4j
+@Testcontainers
+public class CreateNodeE2eTest
+{
+    private static final String BUCKET_NAME = "test-hxinsight-bucket";
+    private static final int INITIAL_DELAY_MS = 200;
+    private static final String PARENT_ID = "-my-";
+    private static final String DUMMY_CONTENT = "Dummy's file dummy content";
+
+    private static final Network network = Network.newNetwork();
+    @Container
+    private static final PostgreSQLContainer<?> postgres = DockerContainers.createPostgresContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> activemq = DockerContainers.createActiveMqContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> sfs = DockerContainers.createSfsContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> transformCore = DockerContainers.createTransformCoreAioContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> transformRouter = DockerContainers.createTransformRouterContainerWithin(network);
+    @Container
+    private static final AlfrescoRepositoryContainer repository = createRepositoryContainer();
+    @Container
+    private static final WireMockContainer hxInsightMock = DockerContainers.createWireMockContainerWithin(network)
+            .withFileSystemBind("src/test/resources/wiremock/hxinsight", "/home/wiremock", BindMode.READ_ONLY);
+    @Container
+    private static final LocalStackContainer awsMock = DockerContainers.createLocalStackContainerWithin(network);
+    @Container
+    private static final GenericContainer<?> liveIngester = createLiveIngesterContainer();
+
+    RepositoryNodesClient repositoryNodesClient = new RepositoryNodesClient(repository.getBaseUrl(), "admin", "admin");
+    AwsS3Client awsS3Client = new AwsS3Client(awsMock.getHost(), awsMock.getFirstMappedPort());
+
+    @BeforeAll
+    @SneakyThrows
+    public static void beforeAll()
+    {
+        WireMock.configureFor(hxInsightMock.getHost(), hxInsightMock.getPort());
+        awsMock.execInContainer("awslocal", "s3api", "create-bucket", "--bucket", BUCKET_NAME);
+    }
+
+    @Test
+    @SneakyThrows
+    void testCreateFile()
+    {
+        // given
+        @Cleanup
+        InputStream fileContent = new ByteArrayInputStream(DUMMY_CONTENT.getBytes());
+        assertThat(awsS3Client.listS3Content()).isEmpty();
+
+        // when
+        Node createdNode = repositoryNodesClient.createNodeWithContent(PARENT_ID, "dummy.txt", fileContent, "text/plaint");
+
+        // then
+        RetryUtils.retryWithBackoff(() -> {
+            assertThat(awsS3Client.listS3Content()).hasSize(1);
+            WireMock.verify(exactly(1), postRequestedFor(urlEqualTo("/presigned-urls")));
+            WireMock.verify(moreThanOrExactly(2), postRequestedFor(urlEqualTo("/ingestion-events"))
+                    .withRequestBody(containing(createdNode.id())));
+        }, INITIAL_DELAY_MS);
```

Comment:
shouldn't we also somehow check if there is some file in s3 and check what it contains?

Reply:
Verification added